### PR TITLE
Fix corrupted WAL complains on standby

### PR DESCRIPTION
**Pull request by arssher**
see commit messages

### CODE REVIEW

File: safekeeper/src/send_wal.rs [hunk 1]
```diff
@@ -566,6 +574,11 @@ impl<IO: AsyncRead + AsyncWrite + Unpin> WalSender<'_, IO> {
                 }))
                 .await?;
 
+            if let Some(appname) = &self.appname {
```
Comment (knizhnik) [inline safekeeper/src/send_wal.rs:577]: Is it added just for testing?
If so I think we should use fail point here...

**Review by knizhnik — CHANGES_REQUESTED** [review]

File: safekeeper/src/send_wal.rs (hunk 1 above)
Comment (arssher) [reply safekeeper/src/send_wal.rs:571]: done

**Review by knizhnik — APPROVED** [review]
It seems to be better to move failpion refactoring in separate PR.
But from my point of view final result is more important than sequence of steps we have made to reach it:)

**Comment by arssher** [comment]
I made separate commits which I think is enough, making separate PRs doesn't give much...

**Pull request merged**